cpu/cpu.go

```diff
@@ -350,8 +350,13 @@ func getStats(path string, stats map[string]interface{}, prevMetricsSum map[stri
 					if err != nil {
 						return err
 					}
-					metricStats[getNamespaceMetricPart(metricName, percentageRepresentationType)] =
-						float64(100 * (currVal - prevVal) / diffSum)
+
+					if percVal := float64(100 * (currVal - prevVal) / diffSum); percVal < 0 {
+						fmt.Fprintln(os.Stderr, "Percentage value could not be calculated due to invalid data reported by /proc/stat")
```

Comment:
Consider reporting also the metric name while indicating problem with the values.
/procfs readings tend to be instantaneous, user might not know what to look for.

Besides that, it's **LGTM**.